hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/DeleteMetadataActor.scala

```diff
@@ -34,6 +43,18 @@ class DeleteMetadataActor(metadataDeletionConfig: MetadataDeletionConfig) extend
       )
       workflowIdsForMetadataDeletionFuture onComplete {
         case Success(workflowIds) =>
+          if (isMetricActorFree) {
```

Comment:
Another idea: what if we make metrics actor maintain his own state var.
We can create a single thread executor inside the metrics actor and submit the `Runnable` like this to that executor:
```
val futureRes =  countRootWorkflowSummaryEntriesByArchiveStatusAndOlderThanTimestamp()
futureRes onComplete {
  case Success =>
    sendGauge()
    self ! FinishedMetricsStuffAndCanAcceptNewRequests
    ...
}
```
If actor's state is `busy` then it will dismiss new requests from deletion actor.
After receiving `FinishedMetricsStuffAndCanAcceptNewRequests` message the metrics actor will update it's state to `free` and will accept new requests from deletion actor again.

@cjllanwarne, @mcovarr how does that sound?

Reply:
Wouldn't that break the state testing below?

I guess I don't understand why the standard FSM pattern wouldn't work well here.

Reply:
FSM would work. I just forgot that `countRootWorkflowSummaryEntriesByArchiveStatusAndOlderThanTimestamp()` is already asynchronous and returns `Future`.